frog/imports/client/Wiki/ModalSettings.js

```diff
@@ -0,0 +1,142 @@
+// @flow
+
+import React, { useState } from 'react';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import TextField from '@material-ui/core/TextField';
+import Switch from '@material-ui/core/Switch';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Divider from '@material-ui/core/Divider';
+import { withStyles } from '@material-ui/styles';
+
+import { Modal } from './components/Modal';
+import { type WikiSettingsT } from './index';
+
+type ModalSettingsPropsT = {
+  callback: WikiSettingsT => void,
+  hideModal: () => void,
+  currentSettings: WikiSettingsT,
+  classes: Object
+};
+
+const styles = {
+  group: {
+    margin: '1vh'
+  },
+  divider: {
+    margin: '2vh'
+  },
+  label: {
+    paddingTop: '2vh'
+  },
+  helperText: {
+    display: 'inline',
+    paddingLeft: '1vw',
+    fontSize: '1rem',
+    letterSpacing: '0.001em'
+  },
+  select: {
+    margin: '1vh'
+  }
+};
+
+export default withStyles(styles)(
+  ({ callback, hideModal, currentSettings, classes }: ModalSettingsPropsT) => {
+    const [settings, setSettings] = useState(currentSettings);
+    const handleChange = target => name => event => {
```

Comment:
I guess it doesn't matter, but why not pass the target, name and event as three props? Isn't this generating a bunch of new functions every time it gets called? (Again I know this won't impact performance here but still)

Reply:
This is syntactically more concise and in our implementation generates the same number of function (1 for handling each change). 
This is because we would need to use arrow functions to pass the props otherwise.